api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/CalculationCohortDefinitionEvaluator.java

```diff
@@ -81,6 +81,14 @@ protected CalculationResultMap doCalculation(
     calcContext.addToCache("onOrAfter", context.getParameterValue("onOrAfter"));
     calcContext.addToCache("onOrBefore", context.getParameterValue("onOrBefore"));
 
+    calcContext.addToCache(
```

Comment:
Those new mappings you have just added to cache, are they NOT just **startDate(onOrAfter)** and **endDate(onOrBefore)** manipulated with a variable integer? Do we really need those added to the evaluator or we can just use those already existing? what are you mapping those to in the report definition UI parameters?

Reply:
`beginPeriodStartDate=${onOrAfter-6m},beginPeriodEndDate=${onOrBefore-6m}," "completionPeriodStartDate=${onOrAfter},completionPeriodEndDate=${onOrBefore+1m}`. We need this to make the parameters available on the Calculation code. We have to improve this in the future I guess. Have this calculation specific parameters passed in a more clean way.

Reply:
@ningosi just moved this logic inside the calculation